### PR TITLE
Color gradient picker

### PR DESCRIPTION
**Pull request by MrOnosa**
Integrated the react-color-gradient-picker from https://www.npmjs.com/package/react-color-gradient-picker

### CODE REVIEW

**Comment by ohlookitsderpy** [comment]
Hello again! Sorry this review is later than usual, I was away.

The circle looks worse than normal now:
(New)
image

(Old)
image

I can't seem to find a way to go back to having just 1 colour on normal colour picker anymore:
image

The text colour on the picker is different to normal as well and the padding is off compared to the rest of the items:
image

Not sure how to show this in a screenshot, but I don't know how to remove the various stops on the picker, or how to remove the initial one (all the gradients I create have a white bit at the bottom/top).

Finally, I think it would be good if we somehow add text to the various functions of the picker as the functions are rather complex to use.

Thank you!

File: src/components/modals/settings/Section.jsx [hunk 1]
```diff
@@ -19,6 +19,10 @@ export default class Section extends React.PureComponent {
       display: display,
       transform: transform
     });
+
+    if(this.props.onToggle) {
```
Comment (ohlookitsderpy) [inline src/components/modals/settings/Section.jsx:23]: Please add a space after "if"

File: src/components/modals/settings/sections/BackgroundSettings.jsx [hunk 2]
```diff
@@ -35,6 +41,23 @@ export default class BackgroundSettings extends React.PureComponent {
     toast(this.props.toastLanguage.reset);
   }
 
+  mapFromGradientSettingsToColorPickerInitalState(gradientSettings) {
```
Comment (ohlookitsderpy) [inline src/components/modals/settings/sections/BackgroundSettings.jsx:44]: This function name is very long

File: src/components/modals/settings/sections/BackgroundSettings.jsx [hunk 3]
```diff
@@ -129,31 +135,56 @@ export default class BackgroundSettings extends React.PureComponent {
     return this.props.language.background.disabled;
   }
 
+  onColorPickerChange = (attrs, name) => {
+    if(process.env.NODE_ENV == 'development'){
```
Comment (ohlookitsderpy) [inline src/components/modals/settings/sections/BackgroundSettings.jsx:139]: space after "if" here as well

File: src/helpers/hexToRgb.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import  rgbToHsv  from './rgbToHsv';
```
Comment (ohlookitsderpy) [inline src/helpers/hexToRgb.js:1]: Imports should only have 1 space after the import statement

File: src/scss/index.scss [hunk 5]
```diff
@@ -100,4 +100,10 @@ body {
 
 ::selection {
     background-color: #c2c2c2;
+}
+
+div.picker-area > div.preview > div.color-hue-alpha > div.alpha,
+div.color-preview-area > div > div:nth-child(5)
+{
```
Comment (ohlookitsderpy) [inline src/scss/index.scss:107]: Please make this consistent with the other css

File: src/components/widgets/Background.jsx [hunk 6]
```diff
@@ -8,7 +8,7 @@ export default class Background extends React.PureComponent {
     if (gradient.length > 1) {
       //Note: Append the gradient for additional browser support.
```
Comment (ohlookitsderpy) [inline src/components/widgets/Background.jsx:9]: It's preferred to have a space after the // but not required

File: src/components/modals/settings/sections/BackgroundSettings.jsx [hunk 7]
```diff
@@ -2,16 +2,22 @@ import React from 'react';
 import { toast } from 'react-toastify';
 import Checkbox from '../Checkbox';
 import Section from '../Section';
+import { ColorPicker } from 'react-color-gradient-picker';
+import 'react-color-gradient-picker/dist/index.css';
+import hexToRgb from '../../../../helpers/hexToRgb'
```
Comment (ohlookitsderpy) [inline src/components/modals/settings/sections/BackgroundSettings.jsx:7]: Missing semicolons

**Review by ohlookitsderpy — CHANGES_REQUESTED** [review]
Just some small code-style tweaks that need doing as well before this can be merged

File: src/scss/index.scss (hunk 5 above)
Comment (MrOnosa) [reply src/scss/index.scss]: I wasn't really sure what you meant. But, please see react-color-picker-gradient-picker-custom-styles.scss for the updates.

**Comment by MrOnosa** [comment]
> I can't seem to find a way to go back to having just 1 colour on normal colour picker anymore:

Use the Reset button.

---

>  I don't know how to remove the various stops on the picker

Select a stop and press the Delete key

---

> or how to remove the initial one

Gradients require at least two stops so we cannot delete beyond that by design. You can drag the initial one around. I am able to get it to start at 0 (the very beginning)

**Review by ohlookitsderpy — APPROVED** [review]

**Comment by ohlookitsderpy** [comment]
Somehow I didn't think of using the delete key and reset button 😂 I was asking about the initial one as I couldn't originally seem to get it to change colour from white, but now I can.


Anyways, thank you! LGTM 👍

**Pull request merged**